### PR TITLE
[CSL-2244] Refactor syncWalletOnImport

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
This PR refactors slightly `syncWalletOnImport` and `syncWalletFromGState`, paving the way to finally restore a wallet asynchronously.

Tests conducted:

- [X] Restoring the genesis wallet from backup works
- [X] Restoring from seed works
- [x] Importing from JSON works

### CODE REVIEW

File: wallet/src/Pos/Wallet/Web/Server/Launcher.hs [hunk 1]
```diff
@@ -65,7 +65,9 @@ walletServer
     => (forall x. m x -> Handler x)
     -> m (Server WalletSwaggerApi)
 walletServer nat = do
-    syncWalletsWithGState =<< mapM findKey =<< myRootAddresses
+    syncResults <- syncWalletsFromGState =<< mapM findKey =<< myRootAddresses
+    -- TODO(adn): Make sure re-throwing potential failures is not disruptive here.
+    mapM_ processSyncResult syncResults
```
Comment (Martoon-00) [inline wallet/src/Pos/Wallet/Web/Server/Launcher.hs]: This one looks duplicate of
https://github.com/input-output-hk/cardano-sl/pull/2459/files#diff-08add64f2bb6347f4c3d0e786086d2b1R86
Dunno how it happened, but we can remove this line completely eventually

Comment (adinapoli-iohk) [reply wallet/src/Pos/Wallet/Web/Server/Launcher.hs]: @Martoon-00 Probably a bad merge on my part, I made a little mess with this branch, I'd have to admit, hehe.

Comment (Martoon-00) [reply wallet/src/Pos/Wallet/Web/Server/Launcher.hs]: Btw, do you actually throw any error there?

Comment (adinapoli-iohk) [reply wallet/src/Pos/Wallet/Web/Server/Launcher.hs]: @Martoon-00 I have decided to just log them (with `logError`), as I'm not sure that throwing an error would be very beneficial as we cannot recover anyway. What do you think?

Comment (Martoon-00) [reply wallet/src/Pos/Wallet/Web/Server/Launcher.hs]: > Probably a bad merge on my part, I made a little mess with this branch, I'd have to admit, hehe.
Nope-nope, it seems to be so even in develop. Need to remember to remove sync here one day.

>  I have decided to just log them (with logError), as I'm not sure that throwing an error would be very beneficial as we cannot recover anyway.
I agree, there is a little we can help with if restore failed. So `processSyncResult` may not even require `MonadThrow` if this makes this piece of code look safer.

Comment (adinapoli-iohk) [reply wallet/src/Pos/Wallet/Web/Server/Launcher.hs]: Good catch. Let me remove that extra constraint!

**Pull request merged**